src/main/webapp/src/app/user-profile/user-profile.component.spec.ts

```diff
@@ -1,24 +1,111 @@
 import {async, ComponentFixture, TestBed} from '@angular/core/testing';
-
+import {RouterModule, ActivatedRoute, convertToParamMap} from '@angular/router';
+import {HttpClient} from '@angular/common/http';
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import {GrowpodUiModule} from '../common/growpod-ui.module';
 import {UserProfileComponent} from './user-profile.component';
+import {User} from '../model/user.model';
 
 describe('UserProfileComponent', () => {
   let component: UserProfileComponent;
   let fixture: ComponentFixture<UserProfileComponent>;
+  let httpTestingController: HttpTestingController;
+
+  // Component-local mocks
+  const MOCK_ARGUMENT = '0';
+  const mockActivatedRoute = {
+    snapshot: {
+      paramMap: convertToParamMap({
+        id: MOCK_ARGUMENT,
+      }),
+    },
+  };
+  const MOCK_OK_RESPONSE: User = {
+    id: MOCK_ARGUMENT,
+    email: 'example@example.com',
+    preferredName: 'User name',
+    biography: 'User biography',
+    zipCode: '12345',
+  };
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule, RouterModule, GrowpodUiModule],
       declarations: [UserProfileComponent],
+      providers: [{provide: ActivatedRoute, useValue: mockActivatedRoute}],
     }).compileComponents();
   }));
 
   beforeEach(() => {
+    // Inject dependencies
+    TestBed.inject(ActivatedRoute);
+    TestBed.inject(HttpClient);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    // Create component
     fixture = TestBed.createComponent(UserProfileComponent);
     component = fixture.componentInstance;
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  /**
+   * Test: Should decode the parameter `id` and make
```

Comment:
nit: Comments aren't needed for `it` blocks as they should self-describing in their first argument: `it('should decode the parameter ... ', () => { ... });` For the rest of the document, both levels of specificity (what is said in the comments and what is said in the block) look OK to me so it's up to you to choose which to keep in the `it` and which to remove.

Reply:
Will change this.